packages/syft/src/syft/service/queue/zmq_queue.py

```diff
@@ -582,43 +645,43 @@ def _run(self):
                                 syft_worker_id=self.syft_worker_id,
                             )
                         except Exception as e:
-                            print(
-                                f"ERROR HANDLING MESSAGE: {e}, {traceback.format_exc()}"
-                            )
+                            logger.exception("Error while handling message. {}", e)
                         finally:
                             self.clear_job()
                     elif command == QueueMsgProtocol.W_HEARTBEAT:
                         pass
                     elif command == QueueMsgProtocol.W_DISCONNECT:
                         self.reconnect_to_producer()
                     else:
-                        print("E: invalid input message: ")
+                        logger.error("Invalid command: {}", command)
                 else:
-                    self.liveness -= 1
-                    if self.liveness == 0:
-                        if self.verbose:
-                            print("W: disconnected from broker - retrying...")
-                        try:
-                            time.sleep(RECONNECT_INTERVAL)
-                        except Exception as e:
-                            print(e, traceback.format_exc())
-                            break
+                    if not self.is_producer_alive():
+                        logger.info("Producer check-alive timed out. Reconnecting.")
                         self.reconnect_to_producer()
```

Comment:
MAYBE: we could keep `self.set_producer_alive()` inside the `reconnect_to_producer` method itself, as we call it in other places

Reply:
I think keeping it decoupled makes it more clear

Reply:
yeah following the pattern here where the one to check for timer resets the timer.